From 7854cd141f6f4c896afdc86229671d443ca7ba10 Mon Sep 17 00:00:00 2001
From: Paul Cercueil <paul@crapouillou.net>
Date: Thu, 18 Feb 2021 22:31:39 +0000
Subject: [PATCH] Add feature macros to more C files

These source files use "struct timespec", which is POSIX 1993.09.

Upstream: Rejected (Upstream does not want these workarounds)
see: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/2493

Signed-off-by: Paul Cercueil <paul@crapouillou.net>
[Retrieved from: https://github.com/swaywm/wlroots/pull/2493]
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Julien: refreshed the patch to be applied with fuzz factor 0]
Signed-off-by: Julien Olivain <ju.o@free.fr>
---
 backend/wayland/output.c                  | 1 +
 types/data_device/wlr_data_device.c       | 1 +
 types/data_device/wlr_drag.c              | 1 +
 types/wlr_export_dmabuf_v1.c              | 1 +
 types/wlr_idle.c                          | 1 +
 types/wlr_keyboard_shortcuts_inhibit_v1.c | 1 +
 types/wlr_pointer_constraints_v1.c        | 1 +
 types/wlr_primary_selection.c             | 1 +
 types/wlr_relative_pointer_v1.c           | 1 +
 types/wlr_screencopy_v1.c                 | 1 +
 types/wlr_virtual_pointer_v1.c            | 1 +
 types/wlr_xdg_decoration_v1.c             | 1 +
 types/xdg_shell/wlr_xdg_popup.c           | 1 +
 types/xdg_shell/wlr_xdg_positioner.c      | 1 +
 types/xdg_shell/wlr_xdg_shell.c           | 1 +
 types/xdg_shell/wlr_xdg_surface.c         | 1 +
 16 files changed, 16 insertions(+)

diff --git a/backend/wayland/output.c b/backend/wayland/output.c
index c4b95e1..d5699d3 100644
--- a/backend/wayland/output.c
+++ b/backend/wayland/output.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdint.h>
 #include <stdio.h>
diff --git a/types/data_device/wlr_data_device.c b/types/data_device/wlr_data_device.c
index 7e67479..2db13a6 100644
--- a/types/data_device/wlr_data_device.c
+++ b/types/data_device/wlr_data_device.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/types/data_device/wlr_drag.c b/types/data_device/wlr_drag.c
index c1fa801..8981ea0 100644
--- a/types/data_device/wlr_drag.c
+++ b/types/data_device/wlr_drag.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/types/wlr_export_dmabuf_v1.c b/types/wlr_export_dmabuf_v1.c
index c1d26e6..cd06615 100644
--- a/types/wlr_export_dmabuf_v1.c
+++ b/types/wlr_export_dmabuf_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <unistd.h>
diff --git a/types/wlr_idle.c b/types/wlr_idle.c
index cb3a1cc..5f1fa95 100644
--- a/types/wlr_idle.c
+++ b/types/wlr_idle.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/types/wlr_keyboard_shortcuts_inhibit_v1.c b/types/wlr_keyboard_shortcuts_inhibit_v1.c
index cc9bac1..3924893 100644
--- a/types/wlr_keyboard_shortcuts_inhibit_v1.c
+++ b/types/wlr_keyboard_shortcuts_inhibit_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <wlr/types/wlr_compositor.h>
diff --git a/types/wlr_pointer_constraints_v1.c b/types/wlr_pointer_constraints_v1.c
index 7a8d209..70b0984 100644
--- a/types/wlr_pointer_constraints_v1.c
+++ b/types/wlr_pointer_constraints_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <limits.h>
 #include <pixman.h>
diff --git a/types/wlr_primary_selection.c b/types/wlr_primary_selection.c
index d576bf0..bac4daa 100644
--- a/types/wlr_primary_selection.c
+++ b/types/wlr_primary_selection.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <wlr/types/wlr_primary_selection.h>
diff --git a/types/wlr_relative_pointer_v1.c b/types/wlr_relative_pointer_v1.c
index 44da791..f6925b8 100644
--- a/types/wlr_relative_pointer_v1.c
+++ b/types/wlr_relative_pointer_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <inttypes.h>
 #include <stdlib.h>
diff --git a/types/wlr_screencopy_v1.c b/types/wlr_screencopy_v1.c
index 4ca659a..76c582e 100644
--- a/types/wlr_screencopy_v1.c
+++ b/types/wlr_screencopy_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <drm_fourcc.h>
diff --git a/types/wlr_virtual_pointer_v1.c b/types/wlr_virtual_pointer_v1.c
index b452ff5..661e613 100644
--- a/types/wlr_virtual_pointer_v1.c
+++ b/types/wlr_virtual_pointer_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <wlr/interfaces/wlr_pointer.h>
diff --git a/types/wlr_xdg_decoration_v1.c b/types/wlr_xdg_decoration_v1.c
index fcd97a5..1850665 100644
--- a/types/wlr_xdg_decoration_v1.c
+++ b/types/wlr_xdg_decoration_v1.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdbool.h>
 #include <stdlib.h>
diff --git a/types/xdg_shell/wlr_xdg_popup.c b/types/xdg_shell/wlr_xdg_popup.c
index fa99acd..8361eef 100644
--- a/types/xdg_shell/wlr_xdg_popup.c
+++ b/types/xdg_shell/wlr_xdg_popup.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
diff --git a/types/xdg_shell/wlr_xdg_positioner.c b/types/xdg_shell/wlr_xdg_positioner.c
index 6a991bb..dae879b 100644
--- a/types/xdg_shell/wlr_xdg_positioner.c
+++ b/types/xdg_shell/wlr_xdg_positioner.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <wlr/util/edges.h>
diff --git a/types/xdg_shell/wlr_xdg_shell.c b/types/xdg_shell/wlr_xdg_shell.c
index 70b777d..16d275a 100644
--- a/types/xdg_shell/wlr_xdg_shell.c
+++ b/types/xdg_shell/wlr_xdg_shell.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include "types/wlr_xdg_shell.h"
diff --git a/types/xdg_shell/wlr_xdg_surface.c b/types/xdg_shell/wlr_xdg_surface.c
index a03b90a..a6267cf 100644
--- a/types/xdg_shell/wlr_xdg_surface.c
+++ b/types/xdg_shell/wlr_xdg_surface.c
@@ -1,3 +1,4 @@
+#define _POSIX_C_SOURCE 199309L
 #include <assert.h>
 #include <stdlib.h>
 #include <string.h>
-- 
2.45.2

